### PR TITLE
Promote AutopilotPassthroughPort feature gate to Alpha

### PR DESCRIPTION
**Pull request by vicentefb**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, press enter to put that in a new line, and remove leading whitespace from that line:
>
> /kind breaking
> /kind bug
> /kind cleanup
> /kind documentation
/kind feature
> /kind hotfix
> /kind release

**What this PR does / Why we need it**:
Promote `AutopilotPassthroughPort` feature gate to Alpha

**Which issue(s) this PR fixes**:
Work towards #3721 

**Special notes for your reviewer**:

### CODE REVIEW

File: build/Makefile [hunk 1]
```diff
@@ -70,7 +70,7 @@ GS_TEST_IMAGE ?= us-docker.pkg.dev/agones-images/examples/simple-game-server:0.3
 BETA_FEATURE_GATES ?= "CountsAndLists=true&DisableResyncOnSDKServer=true"
 
 # Enable all alpha feature gates. Keep in sync with `false` (alpha) entries in pkg/util/runtime/features.go:featureDefaults
-ALPHA_FEATURE_GATES ?= "PlayerAllocationFilter=true&PlayerTracking=true&RollingUpdateFix=true&PortRanges=true&PortPolicyNone=true&Example=true"
+ALPHA_FEATURE_GATES ?= "PlayerAllocationFilter=true&PlayerTracking=true&RollingUpdateFix=true&PortRanges=true&PortPolicyNone=true&AutopilotPassthroughPort=false&Example=true"
```
Comment (zmerlynn) [inline build/Makefile:73]: The instructions might not be totally clear but `ALPHA_FEATURE_GATES` is the inverse of the defaults. So here you way `AutopilotPassthroughPort=true`

File: cloudbuild.yaml [hunk 2]
```diff
@@ -276,7 +276,7 @@ steps:
         declare -A versionsAndRegions=( [1.27]=us-east1 [1.28]=us-west1 [1.29]=europe-west1 )
 
         # Keep in sync with (the inverse of) pkg/util/runtime/features.go:featureDefaults
-        featureWithGate="PlayerAllocationFilter=true&PlayerTracking=true&CountsAndLists=false&RollingUpdateFix=true&PortRanges=true&PortPolicyNone=true&DisableResyncOnSDKServer=false&Example=true"
+        featureWithGate="PlayerAllocationFilter=true&PlayerTracking=true&CountsAndLists=false&RollingUpdateFix=true&PortRanges=true&PortPolicyNone=true&DisableResyncOnSDKServer=false&AutopilotPassthroughPort=true&Example=true"
```
Comment (zmerlynn) [inline cloudbuild.yaml:279]: This one is correct, though. (The instructions in the Makefile should probably use this same language of "(the inverse of)")

File: install/helm/agones/defaultfeaturegates.yaml [hunk 3]
```diff
@@ -25,9 +25,9 @@ PlayerTracking: false
 RollingUpdateFix: false
 PortRanges: false
 PortPolicyNone: false
+AutopilotPassthroughPort: false
```
Comment (zmerlynn) [inline install/helm/agones/defaultfeaturegates.yaml:28]: nit: keep this sorted with the others in the same block

File: pkg/util/runtime/features.go [hunk 4]
```diff
@@ -64,12 +64,12 @@ const (
 	// FeaturePortPolicyNone is a feature flag to allow setting Port Policy to None.
 	FeaturePortPolicyNone Feature = "PortPolicyNone"
 
+	// FeatureAutopilotPassthroughPort is a feature flag that enables/disables Passthrough Port Policy.
+	FeatureAutopilotPassthroughPort Feature = "AutopilotPassthroughPort"
```
Comment (zmerlynn) [inline pkg/util/runtime/features.go:68]: nit: keep this sorted with the others in the same block

File: site/content/en/docs/Guides/feature-stages.md [hunk 5]
```diff
@@ -32,6 +32,7 @@ The current set of `alpha` and `beta` feature gates:
 | [Support for Extended Duration Pods on GKE Autopilot (*1.28+ only*)](https://github.com/googleforgames/agones/issues/3386)  | `GKEAutopilotExtendedDurationPods` | Disabled | `Alpha` | 1.37.0 |
 | [GameServer player capacity filtering on GameServerAllocations](https://github.com/googleforgames/agones/issues/1239)       | `PlayerAllocationFilter`           | Disabled | `Alpha` | 1.14.0 |
 | [Player Tracking]({{< ref "/docs/Guides/player-tracking.md" >}})                                                            | `PlayerTracking`                   | Disabled | `Alpha` | 1.6.0  |
+| [Support Passthrough on GKE Autopilot](https://github.com/googleforgames/agones/issues/3721)                                | `AutopilotPassthroughPort`         | Disabled | `Alpha` | 1.40.0 |
```
Comment (zmerlynn) [inline site/content/en/docs/Guides/feature-stages.md:35]: nit: keep this sorted with the others in the same maturity

File: build/Makefile (hunk 1 above)
Comment (vicentefb) [reply build/Makefile:73]: Ohhhh okok i got confused by this line `Keep in sync with `false` (alpha) entries in pkg/util/runtime/features.go:featureDefaults`, will change!

**Review by zmerlynn — APPROVED** [review]

**Pull request merged**